config/web-app/rbac.yaml

```diff
@@ -0,0 +1,68 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: models-web-app
+  namespace: kubeflow
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: models-web-app-cluster-role
+rules:
+- apiGroups:
+  - authorization.k8s.io
+  resources:
+  - subjectaccessreviews
+  verbs:
+  - create
+- apiGroups:
+  - ""
+  resources:
+  - namespaces
+  - pods
+  - pods/log
+  - events
+  verbs:
+  - get
+  - list
+- apiGroups:
+  - serving.kubeflow.org
+  resources:
+  - inferenceservices
+  - inferenceservices/status
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - delete
+  - deletecollection
+  - patch
+  - update
+- apiGroups:
+  - serving.knative.dev
+  resources:
+  - services
+  - services/status
+  - routes
+  - routes/status
+  - configurations
+  - configurations/status
+  - revisions
+  - revisions/status
+  verbs:
+  - get
+  - list
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: models-web-app-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: models-web-app-cluster-role
+subjects:
+- kind: ServiceAccount
+  name: models-web-app
+  namespace: kubeflow
```

Comment:
ACK, on it

Reply:
In the end I completely removed the namespace here as well, instead of using `kfserving-system`. If I kept the namespace definition in the `subject` then kustomize wouldn't do the substitution and change it to `kubeflow`.

Reply:
Pushed the commit for this, but will wait for an ACK/NAK first before resolving the conversation

Reply:
Interesting how this didn't work as KFServing has other clusterrolebindings with subjects that include the `kfserving-system` namespace that the overlay does successfully overwrite.  Looks like the `namespace: kfserving-system` in the `web-app/kustomization.yaml` is causing this.

Mind removing the namespace from the kustomization.yaml and explicitly adding the `namespace: kfserving-system` to the non-cluster resources under web-app (Deployment, VirtualService, Service, ServiceAccount)? This follows more in line with other config files in KFServing, and you should then also be able to keep the namespace in the subject.

Reply:
@pvaneck you were right on the money! Indeed after:
1. removing the namespace from the `kustomization.yaml` of the web-app
2. explicitly adding the `namespace: kfserving-system` in all the non-cluster resources, and patches

as you suggested worked. Kustomize can now successfully detect and modify the name accordingly. All the namespaces in all the base yamls as well as the patches are pointing to `kfserving-system` and are changed to `kubeflow` when building the corresponding overlay.

Again thanks for your help!

Reply:
Pushed another commit with this latest change